### PR TITLE
Add MXKEncryptionKeysImportView dialog and use it from MXKRoomviewCon…

### PR DESCRIPTION
**Pull request by manuroe**
…troller

### CODE REVIEW

File: MatrixKit/Controllers/MXKRoomViewController.m [hunk 1]
```diff
@@ -3323,21 +3325,79 @@ - (void)showAttachmentInCell:(UITableViewCell*)cell
...
+                         }];
+
+                        [currentAlert addActionWithTitle:[NSBundle mxk_localizedStringForKey:@"attachment_e2e_keys_import"] style:MXKAlertActionStyleDefault handler:^(MXKAlert *alert)
+                         {
+                             if (weakSelf)
+                             {
+                                 typeof(self) self = weakSelf;
+                                 self->currentAlert = nil;
+
+                                 __weak typeof(self) weakSelf2 = self;
```
Comment (giomfo) [inline MatrixKit/Controllers/MXKRoomViewController.m]: Why don't you keep using weakself here?

File: MatrixKit/Controllers/MXKRoomViewController.m [hunk 2]
```diff
@@ -3323,21 +3325,79 @@ - (void)showAttachmentInCell:(UITableViewCell*)cell
...
+
+                                 // Show the keys import dialog
+                                 MXKEncryptionKeysImportView *importView = [[MXKEncryptionKeysImportView alloc] initWithMatrixSession:self->roomDataSource.mxSession];
+                                 currentAlert = importView;
+                                 [importView showInViewController:self toImportKeys:fileURL onComplete:^{
+
+                                     typeof(self) self = weakSelf2;
```
Comment (giomfo) [inline MatrixKit/Controllers/MXKRoomViewController.m]: Reuse here weakself by adding sanity check `if (weakself)`

File: MatrixKit/Views/EncryptionKeys/MXKEncryptionKeysImportView.h [hunk 3]
```diff
@@ -0,0 +1,46 @@
+/*
+ Copyright 2017 OpenMarket Ltd
```
Comment (giomfo) [inline MatrixKit/Views/EncryptionKeys/MXKEncryptionKeysImportView.h]: Add the new copyright...

File: MatrixKit/Views/EncryptionKeys/MXKEncryptionKeysImportView.m [hunk 4]
```diff
@@ -0,0 +1,115 @@
+/*
+ Copyright 2017 OpenMarket Ltd
```
Comment (giomfo) [inline MatrixKit/Views/EncryptionKeys/MXKEncryptionKeysImportView.m]: idem

File: MatrixKit/Views/EncryptionKeys/MXKEncryptionKeysImportView.m [hunk 5]
```diff
@@ -0,0 +1,115 @@
...
+     {
+         if (weakSelf)
+         {
+             typeof(self) self = weakSelf;
+
+             // Retrieve the password
+             UITextField *textField = [self textFieldAtIndex:0];
+             NSString *password = textField.text;
+
+              __weak typeof(self) weakSelf2 = self;
```
Comment (giomfo) [inline MatrixKit/Views/EncryptionKeys/MXKEncryptionKeysImportView.m]: keep using weakself

File: MatrixKit/Controllers/MXKRoomViewController.m (hunk 1 above)
Comment (manuroe) [inline MatrixKit/Controllers/MXKRoomViewController.m]: If we follow that "weak/strong self" design, we need to apply it every time we made an async thing. Objects can be released and lost in the middle on the 2nd async operation. No?

Comment (giomfo) [inline MatrixKit/Controllers/MXKRoomViewController.m]: I don't think but to be honest I'm not sure

**Pull request merged**